cmd/podman/shared/create.go

```diff
@@ -538,18 +538,29 @@ func ParseCreateOpts(ctx context.Context, c *GenericCLIResults, runtime *libpod.
 		}
 	}
 
-	// env overrides any previous variables
-	cmdlineEnv := c.StringSlice("env")
-	if len(cmdlineEnv) > 0 {
-		parsedEnv, err := envLib.ParseSlice(cmdlineEnv)
-		if err != nil {
-			return nil, err
+	if c.IsSet("env") {
+		// env overrides any previous variables
+		cmdlineEnv := c.StringSlice("env")
+		if len(cmdlineEnv) > 0 {
+			parsedEnv, err := envLib.ParseSlice(cmdlineEnv)
+			if err != nil {
+				return nil, err
+			}
+			env = envLib.Join(env, parsedEnv)
 		}
-		env = envLib.Join(env, parsedEnv)
+	}
+
+	labelsList := []string{}
+	if c.IsSet("label") {
+		labelsList = c.StringArray("label")
+	}
+	labelFiles := []string{}
+	if c.IsSet("label-files") {
+		labelFiles = c.StringSlice("label-file")
```

Comment:
I think line 558 should be singular, "label-file"

Reply:
Great find.